superset-frontend/src/explore/controlUtils/getFormDataWithDashboardContext.ts

```diff
@@ -30,19 +30,40 @@ import {
   isFreeFormAdhocFilter,
   isSimpleAdhocFilter,
   NO_TIME_RANGE,
+  isAdhocColumn,
 } from '@superset-ui/core';
+import { Operators } from '../constants';
+import {
+  EXPRESSION_TYPES,
+  translateToSql,
+} from '../components/controls/FilterControl/AdhocFilter';
 
 const simpleFilterToAdhoc = (
   filterClause: QueryObjectFilterClause,
-  clause = 'where',
+  clause: 'WHERE' | 'HAVING' = 'WHERE',
```

Comment:
I noticed we have this `'WHERE' | 'HAVING'` in a few places. It could be nice to bundle them up into a `type AdhocFilterClause = 'WHERE' | 'HAVING'` or similar.

Reply:
👍